### PR TITLE
Adding new target changes

### PR DESCRIPTION
**Pull request by muralinr**

### CODE REVIEW

**Comment by atamazov** [comment]
@junliume This is DRAFT (many required changes are missing)

File: src/solver/batchnorm/backward_spatial_single.cpp [hunk 1]
```diff
@@ -230,7 +230,7 @@ BnBwdTrainingSpatialSingle::GetSolution(const ExecutionContext& context,
 #if WORKAROUND_ISSUE_1146
              && (handle.GetDeviceName() != "gfx90a")
 #endif
-                 )) &&
+             && (handle.GetDeviceName() != "gfx940"))) &&
```
Comment (atamazov) [inline src/solver/batchnorm/backward_spatial_single.cpp:233]: Why this is necessary?

**Comment by atamazov** [comment]
@muralinr Custom tests can't enabled for the new target. Please refer to #1925 (test/CMakeLists.txt changes) and implement support for GFX940_ENABLED/GFX940_DISABLED. Make is disabled by default. Then we'll gradually enable appropriate custom tests by means of GFX940_ENABLED.

**Comment by atamazov** [comment]
@muralinr Please add [MI300] to the title of this PR, thanks.

File: src/solver/batchnorm/backward_spatial_single.cpp (hunk 1 above)
Comment (muralinr) [reply src/solver/batchnorm/backward_spatial_single.cpp:233]: > Why this is necessary?

This is for disabling unsupported batchnorm ASM backward kernel.

**Comment by muralinr** [comment]
> @junliume This is DRAFT (many required changes are missing)

Yes Artem. We want to enable public repository for MI300 testing.  These changes would unblock framework and other teams to start MIOPEN using public repository.

**Comment by muralinr** [comment]
> @muralinr Custom tests can't enabled for the new target. Please refer to #1925 (test/CMakeLists.txt changes) and implement support for GFX940_ENABLED/GFX940_DISABLED. Make is disabled by default. Then we'll gradually enable appropriate custom tests by means of GFX940_ENABLED.

Hi Artem, Can you please do it as you have done #1925 ?  I will be busy with XO bring up and enabling framework teams for performance testing for next few weeks. I want to merge these changes without test/CMakeLists.txt changes and unblock framework and other teams to start using our MIOPEN public repository. Thanks.

File: src/solver/batchnorm/backward_spatial_single.cpp (hunk 1 above)
Comment (atamazov) [reply src/solver/batchnorm/backward_spatial_single.cpp:233]: @muralinr oops, that's simple. Thanks ,resolved.

**Comment by atamazov** [comment]
@muralinr 
> Hi Artem, Can you please do it as you have done #1925 ?

Sure, I will. I am going to enable the same test set as we already have for MI200 (except those that require deprecated solvers, see #2081). Please let me know if this looks good to you.

CC @junliume @JehandadKhan

**Review by atamazov — APPROVED** [review]
LGTM

**Comment by muralinr** [comment]
> @muralinr
> 
> > Hi Artem, Can you please do it as you have done #1925 ?
> 
> Sure, I will. I am going to enable the same test set as we already have for MI200 (except those that require deprecated solvers, see #2081). Please let me know if this looks good to you.
> 
> CC @junliume @JehandadKhan

@atamazov It looks good to me. Thanks.

**Review by junliume — APPROVED** [review]

**Pull request merged**